tests/flytekit/common/workflows/sagemaker.py

```diff
@@ -0,0 +1,122 @@
+import os as _os
+
+from flytekit import configuration as _configuration
+from flytekit.common.tasks.sagemaker import hpo_job_task
+from flytekit.common.tasks.sagemaker.built_in_training_job_task import SdkBuiltinAlgorithmTrainingJobTask
+from flytekit.common.tasks.sagemaker.types import HyperparameterTuningJobConfig
+from flytekit.models.sagemaker.hpo_job import HyperparameterTuningJobConfig as _HyperparameterTuningJobConfig
+from flytekit.models.sagemaker.hpo_job import (
+    HyperparameterTuningObjective,
+    HyperparameterTuningObjectiveType,
+    HyperparameterTuningStrategy,
+    TrainingJobEarlyStoppingType,
+)
+from flytekit.models.sagemaker.parameter_ranges import (
+    ContinuousParameterRange,
+    HyperparameterScalingType,
+    IntegerParameterRange,
+    ParameterRangeOneOf,
+)
+from flytekit.models.sagemaker.training_job import (
+    AlgorithmName,
+    AlgorithmSpecification,
+    InputContentType,
+    InputMode,
+    TrainingJobResourceConfig,
+)
+from flytekit.sdk.types import Types
+from flytekit.sdk.workflow import Input, workflow_class
+
+example_hyperparams = {
+    "base_score": "0.5",
+    "booster": "gbtree",
+    "csv_weights": "0",
+    "dsplit": "row",
+    "grow_policy": "depthwise",
+    "lambda_bias": "0.0",
+    "max_bin": "256",
+    "max_leaves": "0",
+    "normalize_type": "tree",
+    "objective": "reg:linear",
+    "one_drop": "0",
+    "prob_buffer_row": "1.0",
+    "process_type": "default",
+    "rate_drop": "0.0",
+    "refresh_leaf": "1",
+    "sample_type": "uniform",
+    "scale_pos_weight": "1.0",
+    "silent": "0",
+    "sketch_eps": "0.03",
+    "skip_drop": "0.0",
+    "tree_method": "auto",
+    "tweedie_variance_power": "1.5",
+    "updater": "grow_colmaker,prune",
+}
+
+builtin_algorithm_training_job_task2 = SdkBuiltinAlgorithmTrainingJobTask(
+    training_job_resource_config=TrainingJobResourceConfig(
+        instance_type="ml.m4.xlarge", instance_count=1, volume_size_in_gb=25,
+    ),
+    algorithm_specification=AlgorithmSpecification(
+        input_mode=InputMode.FILE,
+        input_content_type=InputContentType.TEXT_CSV,
+        algorithm_name=AlgorithmName.XGBOOST,
+        algorithm_version="0.72",
+    ),
+)
+
```

Comment:
Do you not need to register the tasks first? I took the content of this file and ran it and encountered this error
```
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INTERNAL
	details = "failed to compile workflow for [resource_type:WORKFLOW project:"flyteexamples" domain:"development" name:"SageMakerHPO" version:"with_custom_type1" ] with err entry not found"
	debug_error_string = "{"created":"@1602714149.782035000","description":"Error received from peer ipv6:[::1]:30081","file":"src/core/lib/surface/call.cc","file_line":1062,"grpc_message":"failed to compile workflow for [resource_type:WORKFLOW project:"flyteexamples" domain:"development" name:"SageMakerHPO" version:"with_custom_type1" ] with err entry not found","grpc_status":13}"
>
```
which was fixed after I register the two tasks first.